dcpy/connectors/edm/packaging.py

```diff
@@ -0,0 +1,150 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Callable
+from dataclasses import dataclass
+import typer
+
+from dcpy.utils import s3
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import publishing
+
+BUCKET = "edm-distributions"
+BUCKET_ACL = s3.string_as_acl("public-read")
+
+DOWNLOAD_ROOT_PATH = Path(".library")
+OUTPUT_ROOT_PATH = Path(".output")
+
+
+@dataclass
+class PackageKey(publishing.ProductKey):
```

Comment:
Is this distinct from a PublishKey in any way? Either in usage or intent: a PublishKey is meant to be a unique identifier of a published dataset and version, and it seems like this could just re-use that here

Reply:
Or maybe we need a tweak then, if the "product" in a product key is something like "db-pluto" instead of "pluto". I think it SHOULD cover both use cases, even if it doesn't at this moment. And could be handled differently in different places depending on what's needed

Reply:
I think the difference in folder structure (and so the path property) led me to make PackageKey:
```
class PublishKey(ProductKey):
    ...
    @property
    def path(self) -> str:
        return f"{self.product}/publish/{self.version}"
```
vs
```
class PackageKey(publishing.ProductKey):
    ...
    @property
    def path(self) -> str:
        return f"{self.product}/{self.version}"
```

related to what we were saying earlier thought about publish and package being so similar we may wanna merge them later. I also thought about extracting all ProductKey stuff out of these cli-focused files, but thought adding packing to highlight potential abstraction first would help

Reply:
Gotcha. Yeah I think ideally we'd have one key that really is just that - product name and version. And then it could have like "publish_path" and "package_path" properties. But then we do need more metadata that's really product level. Like Pluto with `name: pluto` and `publishing_folder: db-pluto` etc.

But that's a bit of a broader discussion, and not sure if that's blurring lines a bit with how we've implemented things right now. But something for us to sort out - are these keys meant to map to specific entities like folders in s3/specific build outputs? Or just a version of a product, across multiple locations. Arguments for both, and I think keeping them separate like you have for now probably makes sense